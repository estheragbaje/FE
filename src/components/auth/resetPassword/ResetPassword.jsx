import React, { useState } from 'react';
import { Mutation } from 'react-apollo';
import { Redirect, Link } from 'react-router-dom';
import { toast } from 'react-toastify'; // eslint-disable-line
import ResetPasswordDiv from './StyledResetPassword';
import TextInput from '../../common/TextInput';
import { SEND_RESET_PASSWORD_EMAIL } from '../../../graphql/mutations';
import Button from '../../../styles/Button';

export function ResetPassword() {
  const [email, setEmail] = useState('');
  const [redirectReset, setRedirectReset] = useState(false);

  const onChange = (e) => setEmail(e.target.value);

  if (redirectReset) {
    return <Redirect to="/resetPasswordConfirmation" />;
  }

  return (
    <Mutation mutation={SEND_RESET_PASSWORD_EMAIL}>
      {(sendResetPasswordEmail) => (
        <ResetPasswordDiv>
          <form
            className="text-center p-5"
            action="#!"
            onSubmit={(e) => {
              e.preventDefault();
              sendResetPasswordEmail({
                variables: {
                  email,
                },
              })
                .then(() => {
                  setRedirectReset(true);
                })
                .catch(() => {
                  toast('Failed to send reset email password', {
                    className: 'toast-error',
                  });
                })
                .finally(() => setEmail(''));
            }}
          >
            <p className="h4 mb-4 f-1">Reset Password</p>
            <p>
              Enter your email in the input below and we will help you recover
              your password.
            </p>

            <TextInput
              title="Email"
              name="email"
              type="email"
              value={email}
              onChange={onChange}
              id="email"
              className="form-control mb-4"
              required
            />

            <Button className="btn btn-block my-4" type="submit">
              Reset Password
            </Button>

            <p>
              Back to &nbsp;
              <Link to="/login">
                <u>Sign In</u>
              </Link>
            </p>
          </form>
        </ResetPasswordDiv>
      )}
    </Mutation>
  );
}

export default ResetPassword;
