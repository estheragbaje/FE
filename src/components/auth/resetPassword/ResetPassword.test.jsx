import React from 'react';
import { shallow } from 'enzyme';

import ResetPassword from './ResetPassword';

describe('<ResetPassword />', () => {
  it('renders the ResetPassword component correctly', () => {
    const wrapper = shallow(<ResetPassword />);
    expect(wrapper).toMatchSnapshot();
  });
});
