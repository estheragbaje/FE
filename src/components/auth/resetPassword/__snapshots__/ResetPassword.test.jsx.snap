// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword /> renders the ResetPassword component correctly 1`] = `ShallowWrapper {}`;
